### DISCUSSION TITLE
Anyone moved over from Hugo yet? Challenges, suggestions, surprises?

### DISCUSSION THREAD

**Discussion by danielfdickinson** [bridgetownrb/bridgetown#798] [General]
I ran across bridgetown after reading [a Spicy web post pointed to by Bryce Wray, met via Hugo](https://www.brycewray.com/posts/2023/08/mixed-nuts-12/) and am considering trying it out. Since I just got started on [a new theme that currently is just scaffolding](https://github.com/wildtechgarden/a-wild-theme-mod-hugo), now would be a good time to try a different SSG.

So, in building a theme (or converting an existing site/theme) when one is used to Hugo, are there any particular challenges, suggestions, or surprises?

<details>
<summary>Aside</summary>
<div>
<p>I'm a bit extra curious about Bridgetown because the first static site generator I used (long before Jekyll, Hugo, or JAMstack), was a Ruby-based generator call [webgen](https://github.com/gettalong/webgen), and Ruby seems to keep coming up (I was skeptical of Ruby when it was first introduced, TBH, because it seemed like there were a bunch of 'Kewl new' scripting (and other) languages, and I was reluctant to pick one or the other as winners/losers. Python, Go, and Rust (although it got off to slow start, as I recall) also were getting quite a bit of promotion then).<p>

<p>I'm not a website designer or frontend developer by trade, but I've never been particularly happy about the commercial offerings for building websites and have was an early adopter of static, no or minimal JavaScript websites (for sites vs. apps; I never quite understood why information sites needed a great deal of JavaScript, except to do local search).</p>
</div>
</details>

I've spent a lot of time learning the ins and outs of Hugo, but have never really liked the Go templates, or the lack of ability to do testing of the templates. With bridgetown, are the templates something that can be run through automated testing?

Any bonus of things possible with Bridgetown that are not possible with Hugo?

**Comment by danielfdickinson** [comment]
So, it looks like Bridgetown if more JS and web-app vs. static website oriented than I thought. (I've been browsing around the docs, discussions, and issues). I think I'll likely stick with Hugo for now, since I'd rather not have a JavaScript stack (except for doing search, and limiting form spam, and the latter I have to think about more, yet), and am still debating vanilla CSS vs. SASS for my new theme.

The Bridgetown / Spicy Web philosophy of rethinking the modern web I think is on the right track, but for basic websites, I don't think it goes far enough away from the 'heavy' (so-called modern) web.

I'll keep this open, though, in case I'm wrong, or you want to discuss further.

**Comment by jaredcwhite** [comment]
@danielfdickinson Hey, thanks for checking this out. I'm wondering if when you say "I'd rather not have a JavaScript stack" if you mean you'd rather not use JavaScript on your static site, or you'd rather not have Node installed locally? Because Bridgetown definitely supports the former (zero kb JS shipping is easy-peasy), whereas it _does_ require Node to support bundling CSS/Sass (and JS if you do end up wanting some). Generally speaking any website development these days in any framework pretty much requires a local Node install unless you're doing something suuuper basic.

**Comment by danielfdickinson** [comment]
@jaredcwhite I'd say I've got super basic needs (what you have called the content-rich document web, like https://www.wildtechgarden.ca). Just blogs and documentation sites at present. Other than local search I don't need JS, and for CSS/Sass, Hugo does the processing, so I don't need Node for that. My preliminary work on a new theme 'shell' goes so far as to vanilla CSS that I have coded by hand (it's not great CSS but introducing pipelines just to handle that would be a bit much).

I think what you are doing here makes sense for a web apps and sites other than the kind I build.

I'm also a bit of an 'unfan' of the Node ecosystem in general. The number of dependencies is a 'tad' excessive in my opinion.

Anyway, sorry for ranting in your space. This does look like an interesting project, just not for me at this time.
